base/set.jl

```diff
@@ -54,8 +54,21 @@ function union(s::Set, sets::Set...)
     return u
 end
 const ∪ = union
+
 union!(s::Set, xs) = (for x=xs; push!(s,x); end; s)
 union!(s::Set, xs::AbstractArray) = (sizehint!(s,length(xs));for x=xs; push!(s,x); end; s)
+for T in [Bool,Int8,UInt8]
+    MAXSIZE = Int(typemax(T))-Int(typemin(T))+1
+    @eval begin
+        function union!(s::Set{$T}, xs::AbstractArray{$T})
+            while i<=length(xs) && length(s) != $MAXSIZE
```

Comment:
Is `i` actually undefined here? Anyway, if you want this to be fast for all array types, use `eachindex` in addition to a counter.

This also means the tests do not cover this part...